packages/suite/src/middlewares/wallet/fiatRatesMiddleware.ts

```diff
@@ -51,23 +51,19 @@ const fiatRatesMiddleware = (api: MiddlewareAPI<Dispatch, AppState>) => (next: D
         case ACCOUNT.CREATE: {
             // fetch current rates for account's tokens
             const account = action.payload;
-            if (account.tokens) {
-                account.tokens.forEach((t, i) => {
-                    if (t.symbol) {
-                        const s = t.symbol;
-                        // wait 500ms before firing next fetch
-                        setTimeout(() => {
-                            api.dispatch(
-                                fiatRatesActions.updateCurrentRates({
-                                    symbol: s,
-                                    mainNetworkSymbol: account.symbol,
-                                    tokenAddress: t.address,
-                                }),
-                            );
-                        }, i * 500);
-                    }
-                });
-            }
+            account.tokens?.forEach(t => {
+                if (!t.symbol) {
+                    return;
+                }
```

Comment:
Could be nice to this filtering in an actual `filter` function

```suggestion
account.tokens?
    .filter(t => t.symbol)
    .forEach(t => {
```

Afaik the typescript can infer that t.symbol cannot be `undefined` in the `forEach` block.

Reply:
> Afaik the typescript can infer that t.symbol cannot be `undefined` in the `forEach` block.

it really can't

Reply:
Ah sorry. It really can't *on itself*. (I was expecting otherwise). There still is a way to do this with a help of type-guards:
```typescript
consr arr = [1, 2, 3, undefined]

arr
    .filter((it): it is number => it !== undefined)
    .forEach(it => { /* `it` is inferred to be number here */ })
```

However I found no wat to apply a typeguard to this particular case (on a nested property) and it would be combersome anyway. Early return is best here. Resolving.